File: LeetCode Solutions/December LeetCoding Challenge/Validate Binary Search Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.last = None
    def isValidBST(self, root: TreeNode) -> bool:
        if(not root):
            return True
        if(not self.isValidBST(root.left)):
            return False
        if(self.last is not None and self.last >= root.val):
            return False
        self.last = root.val
        return self.isValidBST(root.right)

File: LeetCode Solutions/Top Interview Questions(Easy Collection)/Trees/Validate Binary Search Tree/Validate Binary Search Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isValidBST(TreeNode root) {
        return isValidBST(root,Long.MAX_VALUE,Long.MIN_VALUE);
    }
    public boolean isValidBST(TreeNode root,long min,long max){
        if(root == null)
            return true;
        if(root.val >= min || root.val <= max)
            return false;
        return isValidBST(root.left,root.val,max) && isValidBST(root.right,min,root.val);
    }
}